documents/coredocument.go

```diff
@@ -70,6 +70,15 @@ func compactProperties(key string) []byte {
 
 // CoreDocument is a wrapper for CoreDocument Protobuf.
 type CoreDocument struct {
+	// DataModified indicates that the data of the model that embeds the CoreDocument has been modified and salts needs to be generated for new fields in data precise-proof tree.
+	DataModified bool
```

Comment:
I think we don't need three different flags here. There is only one salts field anyway.